### DISCUSSION TITLE
unique column id apart of blocking rules/comparisons (settings)

### DISCUSSION THREAD

**Discussion by jkginfinite** [moj-analytical-services/splink#1492] [Q&A] [answered]
What is the relationship between the unique column id and blocking rules or the comparisons in settings?

Should the unique column id be included in the comparisons section of the settings?

Should the unique column id be included in any of the blocking rules?

**Comment by ADBond** [comment, marked as answer]
You will almost never want to include the unique column id in comparisons or blocking rules.

The unique column id is a column in your dataset which uniquely identifies each _record_ in your data. This is what Splink uses to identify specific records - so that we can keep track of which records we have paired against which, match up with any labelled data, etc. As there will in general be several _records_ corresponding to the same _person_ (assuming you are linking people), when we are looking at two records corresponding to the same person, many other columns may match (maybe even all of them!), so we need a way to distinguish record 'A' of person '1' from record 'B' of person '1'. This is what the unique column id does.

Almost always this unique column id will be some arbitrary number assigned to each record, rather than having any underlying meaning, and arbitrary numbers generally are not useful for linking, because they provide no information about the underlying person.

Note that _by design_ you will never have a pair of records that has an _exact match_ on unique column id, as no two records can have the same value.

If for some unusual reason you _do_ happen to have a unique column id which has some actual inherent meaning you _might_ consider including it in a comparison (though of course you wouldn't include an [`exact_match_level`](https://moj-analytical-services.github.io/splink/comparison_level_library.html#splink.comparison_level_library.ExactMatchLevelBase)), but you probably still wouldn't use them in blocking rules, as you [almost always want to use exact-match-like conditions in your blocking rules](https://moj-analytical-services.github.io/splink/topic_guides/blocking/performance.html?h=equi#equi-join-conditions). However, even if you _do_ find yourself in this situation it is highly likely that a better idea would be to make a _new_ column that extracts whatever semantically meaningful information from your unique id, and include _that_ in your comparison instead - your model will be clearer, potentially more performant, and you will be able to better leverage [pre-built comparisons](https://moj-analytical-services.github.io/splink/comparison_library.html). I'm also not 100% certain whether Splink would even operate properly if you include the `unique_column_id` in a comparison.

**Reply by jkginfinite** [reply]
great answer thankyou team
